### PR TITLE
Update samples instead of insert and delete

### PR DESCRIPTION
**Pull request by mrdadah**

### CODE REVIEW

**Comment by seveibar** [comment]
I'm going to modify this branch to fix the prettier configuration and rerun the prettier so the diffs are easier to read. I forgot to mention that every project should have a `.prettierrc` with the value `{ "semi": false }`. I've added that to the https://sp.workaround.ai/prettier

File: src/db/schema.js [hunk 1]
```diff
@@ -28,27 +28,15 @@ module.exports = (db) => {
         CREATE TABLE IF NOT EXISTS sample_state (
             sample_state_id INTEGER PRIMARY KEY AUTOINCREMENT,
             session_short_id TEXT NOT NULL,   
-            session_sample_id INTEGER NOT NULL,     
-            version INTEGER NOT NULL DEFAULT 0,
+            session_sample_index INTEGER NOT NULL,     
+            summary_version INTEGER NOT NULL DEFAULT 0,
```
Comment (seveibar) [inline src/db/schema.js:32]: nice

File: src/endpoints/get-diffs.js [hunk 2]
```diff
@@ -3,8 +3,9 @@ const cors = require("micro-cors")()
 const query = require("micro-query")
 const hash = require("../../utils/hash.js")
 const error = require("../../utils/error")
-const getDB = require('../db')
-const db = getDB({databasePath: 'udt.db', verbose: null})
+const getSessionUDT = require("../../utils/getSessionUDT")
```
Comment (seveibar) [inline src/endpoints/get-diffs.js:6]: nice

File: src/endpoints/get-diffs.js [hunk 3]
```diff
@@ -17,33 +18,40 @@ module.exports = cors(async (req, res) => {
...
+    )
+    .all(session.short_id)
 
   const samples = []
-  samplesQueryResults.forEach(sample => {
-    const content = JSON.parse(sample.content)
-    const annotation = JSON.parse(sample.annotation)
-    samples.push(Object.assign({}, content, {annotation}))
+  samplesQueryResults.forEach((sample) => {
+    const sessionUDT = getSessionUDT(sample)
```
Comment (seveibar) [inline src/endpoints/get-diffs.js:38]: hmm this confuses me. why would you call `getSessionUDT` with a sample...

File: utils/getSessionUDT.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+const getSessionUDT = (sample) => {
+  const content = JSON.parse(sample.content);
+  const annotation = JSON.parse(sample.annotation);
+  return { ...content, annotation };
+};
```
Comment (seveibar) [inline utils/getSessionUDT.js:5]: function name isn't good, it's not returning a "UDT JSON Object" (AKA "Dataset Object"), it's return a "Sample Object". Rename to `getSampleObject`.

File: src/endpoints/patch-session.js [hunk 5]
```diff
@@ -1,174 +1,202 @@
...
+
+  const samples = []
+  samplesQueryResults.forEach((sample) => {
+    const sessionUDT = getSessionUDT(sample)
+    samples.push(sessionUDT)
+  })
+
+  newJSON.samples = samples
+
+  return send(res, 200, { latestVersion, hashOfLatestState: hash(newJSON) })
```
Comment (seveibar) [inline src/endpoints/patch-session.js:67]: this implementation looks correct to me!!! nice :+1:

File: utils/getSessionUDT.js (hunk 4 above)
Comment (mrdadah) [reply utils/getSessionUDT.js:5]: done

File: src/endpoints/get-diffs.js (hunk 3 above)
Comment (mrdadah) [reply src/endpoints/get-diffs.js:38]: renamed to `getSampleObject`

**Pull request merged**